utils/threshold/threshold_test.go

```diff
@@ -1,31 +1,116 @@
 package threshold
 
 import (
-	"testing"
-
 	"github.com/herumi/bls-eth-go-binary/bls"
 	"github.com/stretchr/testify/require"
+	"testing"
 )
 
 func TestSplitAndReconstruct(t *testing.T) {
 	Init()
+	sharesCount := uint64(4)
+	threshold := uint64(3)
+	message := []byte("bloxRocks!")
 
 	// generate random secret and split
 	sk := bls.SecretKey{}
 	sk.SetByCSPRNG()
-	shares, err := Create(sk.Serialize(), 4)
+
+	originalSig := sk.SignByte(message)
+
+	shares, err := Create(sk.Serialize(), threshold, sharesCount)
 	require.NoError(t, err)
 
 	// partial sigs
-	m := make(map[uint64][]byte)
+	sigVec := make(map[uint64][]byte)
 	for i, s := range shares {
-		partialSig := s.SignByte([]byte("hello"))
-		m[i] = partialSig.Serialize()
+		partialSig := s.SignByte(message)
+		sigVec[i] = partialSig.Serialize()
 	}
 
 	// reconstruct
-	sig, _ := ReconstructSignatures(m)
-	originalSig := bls.Sign{}
+	sig, _ := ReconstructSignatures(sigVec)
+	require.True(t, originalSig.IsEqual(sig))
 	require.NoError(t, originalSig.Deserialize(sig.Serialize()))
-	require.True(t, originalSig.VerifyByte(sk.GetPublicKey(), []byte("hello")))
+	require.True(t, originalSig.VerifyByte(sk.GetPublicKey(), message))
```

Comment:
@lior-blox we should add a test for invalid partial sigs to make sure it's actually working.

Reply:
@alonmuroch done